### PR TITLE
polymc: new, 1.2.0

### PR DESCRIPTION
**Pull request by OriginCode**
Topic Description
-----------------

Add package `polymc`

Package(s) Affected
-------------------

`polymc` 1.1.1

Security Update?
----------------

No

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

### CODE REVIEW

**Comment by OriginCode** [comment]
FTBFS on amd64:
圖片

**Comment by OriginCode** [comment]
@OriginCode @liushuyu

**Comment by OriginCode** [comment]
> FTBFS on amd64: 圖片

Solved by disabling LTO.

**Comment by OriginCode** [comment]
LGTM on amd64:
圖片

File: extra-games/polymc/autobuild/prepare [hunk 1]
```diff
@@ -0,0 +1,2 @@
+acbs_copy_git
+git submodule update --init --recursive
```
Comment (MingcongBai) [inline extra-games/polymc/autobuild/prepare]: This line is unnecessary.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-games/polymc/autobuild/prepare (hunk 1 above)
Comment (OriginCode) [reply extra-games/polymc/autobuild/prepare]: Line 2?

Comment (WhiredPlanck) [reply extra-games/polymc/autobuild/prepare]: > Line 2?

Yep, ACBS will do it automatically.

**Review by cthbleachbit — CHANGES_REQUESTED** [review]
Changes SUGGESTED:

- We might as well reuse the API key we currently have from multimc5 instead of hardcoded upstream API key. Specify via CMake option` -DLauncher_MSA_CLIENT_ID`.
- Set `Launcher_VERSION_BUILD` to `PKGREL`

Changes REQUESTED:

- PolyMC currently installs two jars to `/usr/share/jar`. We can move them to `/usr/lib/polymc` or `/usr/share/polymc`. Might need to patch `CmakeLists.txt` since the path is hardcoded at line 206.

**Review by cthbleachbit — DISMISSED** [review]
Changed made myself. Waiting for feedback from others.

**Comment by OriginCode** [comment]
1.2.0 LGTM on amd64:
圖片

File: extra-games/polymc/spec [hunk 2]
```diff
@@ -0,0 +1,4 @@
+VER=1.2.0
+SRCS="git::commit=tags/$VER::https://github.com/PolyMC/PolyMC"
```
Comment (liushuyu) [inline extra-games/polymc/spec:2]:
```suggestion
SRCS="git::commit=tags/$VER;copy-repo=true::https://github.com/PolyMC/PolyMC"
```

File: extra-games/polymc/autobuild/prepare [hunk 3]
```diff
@@ -0,0 +1 @@
+acbs_copy_git
```
Comment (liushuyu) [inline extra-games/polymc/autobuild/prepare]:
```suggestion

```

**Review by liushuyu — COMMENTED** [review]
Use https://github.com/PolyMC/libnbtplusplus/commit/129be45a7f91920e76673af104534d215c497d85 to resolve the LTO issue.

File: extra-games/polymc/spec (hunk 2 above)
Comment (OriginCode) [reply extra-games/polymc/spec:2]: LTO issue has been resolved by upgrading PolyMC to 1.2.0.

File: extra-games/polymc/autobuild/defines [hunk 4]
```diff
@@ -0,0 +1,11 @@
+PKGNAME=polymc
+PKGSEC=games
+PKGDEP="openjdk qt-5 x11-app"
+PKGSUG="openjdk-11 openjdk-8"
+BUILDDEP="cmake zlib mesa"
+PKGDES="A free, open source launcher for Minecraft"
+
+CMAKE_AFTER="
+	-DLauncher_PORTABLE=OFF
+	-DLauncher_MSA_CLIENT_ID=81a207c0-a53c-46a3-be07-57d2b28c1643
```
Comment (MingcongBai) [inline extra-games/polymc/autobuild/defines]: Record in comment where this came from.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Review by cthbleachbit — DISMISSED** [review]
Tested to work on amd64

File: extra-games/polymc/autobuild/defines (hunk 4 above)
Comment (OriginCode) [reply extra-games/polymc/autobuild/defines]: `# "AOSC MC" Microsoft APPID for logging in with Microsoft accounts`

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**